Cargo.lock

```diff
@@ -924,6 +935,19 @@ dependencies = [
  "syn",
 ]
 
+[[package]]
+name = "diaglint"
+version = "0.1.3"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "6c1144248ed7ef2fc01b2e650877e085d651530dead491da0a088725138a5b5a"
+dependencies = [
+ "codespan-reporting",
```

Comment:
I see that `diaglint` uses codespan-reporting, which might be fine, but we already have _two_ error reporting libs in the codebase. I'd prefer us to reuse one of them (likely annotate-snippets).

Reply:
Switched to `annotate-snippets`. ec0a6ee3127b0a005807e919676aa578aa0a93f1